### PR TITLE
Add dynamic announcements

### PR DESCRIPTION
**Pull request by jirassimok**

### CODE REVIEW

**Review by cawallin — CHANGES_REQUESTED** [review]
Nice job getting dynamic announcements working!

File: presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/DiscoveryAgentMap.java [hunk 1]
```diff
@@ -122,7 +121,7 @@ private synchronized void refreshNodes()
                         String.format("Invalid URI '%s' for node with ID '%s'",
                                 properties.get("http"), id), e);
             }
-            Agent agent = new Agent(uri, isCoordinator, isWorker, id);
+            Agent agent = new Agent(uri, isCoordinator, id);
```
[the remark this answers is not in the document]
Comment (jirassimok) [reply presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/DiscoveryAgentMap.java]: That's not currently included in the dynamic announcement, but shouldn't be too hard to add. Until then, I'd rather remove it, so it doesn't _look_ like we support that.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/announcement/DynamicAnnouncementBinder.java [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Binder for adding dynamic properties to announcements.
+ * <p>
+ * Must be used alongside DynamicAnnouncementModule.
+ * <p>
+ * Acts as an extension to the {@link Binder} EDSL.
+ * <pre>{@code
+ * dynamicAnnouncementBinder(binder)
```
[the remark this answers is not in the document]
Comment (jirassimok) [reply presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/announcement/DynamicAnnouncementBinder.java]: All right.

**Comment by jirassimok** [comment]
I've fixed the typo that was causing the dynamic announcement to be created incorrectly (the property name was wrong).

I've also added a new commit making the Controller's methods use `String` instead of `UUID` for node IDs, because Airlift's `ServiceDescriptor` (the basis for the discovery announcements) uses `String`.

This should be tested on multiple Agents and with proper configuration before it is merged, but I believe it should work.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/announcement/DynamicAnnouncementBinder.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jirassimok) [reply presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/announcement/DynamicAnnouncementBinder.java]: All right.

**Comment by cawallin** [comment]
Merged, thanks!

**Pull request merged**